services/galley/src/Galley/API/Action.hs

```diff
@@ -442,17 +443,25 @@ performConversationJoin qusr lconv (ConversationJoin invited role) = do
       Local UserId ->
       Sem r ()
     checkRemoteBackendsConnected lusr = do
-      let invitedDomains = tDomain <$> snd (partitionQualified lusr $ NE.toList invited)
-          existingDomains = tDomain . rmId <$> convRemoteMembers (tUnqualified lconv)
-
-      -- Note:
-      --
-      -- In some cases, this federation status check might be redundant (for
-      -- example if there are only local users in the conversation). However,
-      -- it is important that we attempt to connect to the backends of the new
-      -- users here, because that results in the correct error when those
-      -- backends are not reachable.
-      checkFederationStatus (RemoteDomains . Set.fromList $ invitedDomains <> existingDomains)
+      let remoteUsers = snd (partitionQualified lusr $ NE.toList invited)
+          invitedRemoteDomains = Set.fromList . filter (/= tDomain lconv) $ tDomain <$> remoteUsers
+          existingRemoteDomains = Set.fromList $ tDomain . rmId <$> convRemoteMembers (tUnqualified lconv)
+          allInvitedAlreadyInConversation = null $ invitedRemoteDomains \\ existingRemoteDomains
+
+      if not allInvitedAlreadyInConversation
+        then -- Note:
+        --
+        -- In some cases, this federation status check might be redundant (for
+        -- example if there are only local users in the conversation). However,
+        -- it is important that we attempt to connect to the backends of the new
+        -- users here, because that results in the correct error when those
+        -- backends are not reachable.
+          checkFederationStatus (RemoteDomains (invitedRemoteDomains <> existingRemoteDomains))
+        else do
+          -- even if there are no new remotes, we still need to check they are reachable
+          void . (ensureNoUnreachableBackends =<<) $
+            E.runFederatedConcurrentlyEither remoteUsers $ \_ ->
+              void $ fedClient @'Brig @"api-version" ()
```

Comment:
This is weird, because it will result in 2 calls to api-version in the success case. But I guess this "ping" logic already exists elsewhere. Can't it be fixed simply by having `pure ()` as the body of `runFederatedConcurrentlyEither`?